### PR TITLE
doc: remove VBSK related content

### PR DESCRIPTION
**Pull request by liuhang-bit**
VBSK is not supported anymore, clean the documents up.

Track-On: #6738
Signed-off-by: hangliu1 <hang1.liu@linux.intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/hld-virtio-devices.rst [hunk 1]
```diff
@@ -547,8 +448,7 @@ VBS APIs
 ========
 
 The VBS APIs are exported by VBS related modules, including VBS, DM, and
-Service VM kernel modules. They can be classified into VBS-U and VBS-K APIs
-listed as follows.
+Service VM kernel modules.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-virtio-devices.rst:451]: Are there still kernel modules providing VBS APIs now that we do not have VBS-K anymore?

File: doc/user-guides/acrn-dm-parameters.rst [hunk 2]
```diff
@@ -601,8 +601,7 @@ arguments used for configuration.  Here is a table describing these emulated dev
          a FE GPIO, you can set a new name here.
 
    * - ``virtio-rnd``
-     - Virtio random generater type device, with string ``kernel=on`` to
-       select the VBSK virtio backend. The VBSU virtio backend is used by default.
+     - Virtio random generater type device, the VBSU virtio backend is used by default.
```
Comment (gvancuts) [inline doc/user-guides/acrn-dm-parameters.rst:604]:
```suggestion
     - Virtio random generator type device, the VBSU virtio backend is used by default.
```

**Comment by wenlingz** [comment]
@liuhang-bit make doc fail, please check:
New errors/warnings found, please fix them:
==============================================

/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/contribute.rst:20: WARNING: toctree contains reference to nonexisting document 'developer-guides/VBSK-analysis'
/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/release_notes/release_notes_2.3.rst:77: WARNING: undefined label: vbsk-overhead (if the link has no caption the label must precede a section header)

Makefile:61: recipe for target 'html' failed
make[1]: *** [html] Error 1
make[1]: Leaving directory '/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/doc'
Makefile:137: recipe for target 'doc' failed
make: *** [doc] Error 2

**Comment by wenlingz** [comment]
@liuhang-bit make doc fail

New errors/warnings found, please fix them:
==============================================

/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/release_notes/release_notes_2.3.rst:77: WARNING: undefined label: vbsk-overhead (if the link has no caption the label must precede a section header)

Makefile:61: recipe for target 'html' failed
make[1]: *** [html] Error 1
make[1]: Leaving directory '/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/doc'
Makefile:137: recipe for target 'doc' failed
make: *** [doc] Error 2
+ error make doc
++ timestamp
++ date +%T
+ echo '10:05:49 ERROR: make doc'
10:05:49 ERROR: make doc

**Comment by dbkinder** [comment]
@liuhang-bit I merged #6864 that removes the problem where removing a document broke doc generation of the 2.3 release notes

**Comment by dbkinder** [comment]
ok to verify

**Comment by wenlingz** [comment]
OK to verify doc

**Comment by dbkinder** [comment]
@liuhang-bit I suspect you need to rebase to pick up my PR that fixes this doc build error

**Review by dbkinder — APPROVED** [review]

**Pull request merged**